sklearn/discriminant_analysis.py

```diff
@@ -314,6 +316,16 @@ class LinearDiscriminantAnalysis(
     [1]
     """
 
+    _parameter_constraints = {
+        "solver": [StrOptions({"svd", "lsqr", "eigen"})],
+        "shrinkage": [StrOptions({"auto"}), Interval(Real, 0, 1, closed="both"), None],
+        "n_components": [Interval(Integral, 1, None, closed="left"), None],
+        "priors": ["array-like", None],
+        "store_covariance": [bool],
+        "tol": [Interval(Real, 0, None, closed="left")],
+        "covariance_estimator": [HasFit, None],
```

Comment:
```suggestion
        "covariance_estimator": "no_validation",
```

Reply:
I have made the changes as per your suggestions

Reply:
Passing `no_validation` in `covariance_estimator` causes tests to fail.